### ISSUE TITLE
Meeting series scoped

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/parabol#4584]
>When team leads groom the calendar for the team, setting up their agile meetings, they want a recurring meeting space they can link to so that everybody gets to the right spot.

Meeting series aims to unlock this.

## Drafted notes are deprecated, iterative thinking
[See proposal in comment below](https://github.com/ParabolInc/parabol/issues/4584#issuecomment-772705804)

#### Basics
- All meetings are a series, some are just a series of 1
- A meeting series creates the next meeting object when the previous meeting is completed
- A permalink to the series is a link to the current meeting object that is open. This should be useful for folks who manually add a link to the meeting on their future calendar events

#### Editing the name
- The meeting series gets a default name ‘Retro 1’
  - Each meeting will increment the name until the name is edited ‘Retro 1, Retro 2, Retro 3’
  - If they edit the name e.g. ‘Product Sprint’ then that name persists across future meetings in the series
  - They can update the meeting name at any time from the meeting card or the meeting interface
  - When a meeting is completed, the summary uses the name as it was for that meeting

#### A simple option to open the next meeting
- We can take an easy first step today. [We can add a checkbox that opens the next meeting](https://www.figma.com/file/dfTQxhoKzohkCpxlztBbhS/Meetings-view?node-id=121%3A70482). Until the meeting series is canceled from the meetings view, closing the meeting will open the next one.

#### Scheduling a single date, time
- Folks can add a date and a time as meta on the meeting object. This may be great for ad hoc meetings series of 1. This could be tedious for folks who want a recurring pattern.

#### Recurring patterns
- Recurring patterns can be added and edited. Any changes in the recurring pattern apply to the current and future meeting objects. E.g. change ‘every Monday’ to ‘every Tuesday’ will apply to a currently open meeting, it will have a date that says the next Tuesday.
- We use recurring patterns like Google Calendar for day, week, month
- We set them up to be recurring by default unless they drill down and change it.
  - We pick a drunken default: Retro every 2 wks, Poker every 2 wks, Check-In every 1 week, Daily Scrum…
- The next meeting object in a series can be blocked by an administrative step before folks can participate in the first phases e.g. retros => pick your template! => icebreaker/reflect
- The meeting series shows up on the meetings view
- The meeting series is created from the new meeting view
- The series can be edited from the meeting card (name, scheduling)
- If folks change the date but not the recurring pattern, the date only changes for the open meeting
  - The next meeting scheduling could be changed to adjust for an upcoming holiday

#### Pushing to a calendar
- Folks will be prompted to integrate with their calendar
- Meetings can have a single date or recurring pattern that can be pushed to their calendar. This is a 1-way push
- To keep it simple we can push 1 meeting event at a time. There they can groom the event, add folks, etc. Basically they have a shortcut to creating an event that points back to the Parabol meeting space
- If folks edit an existing series and update the date/time and/or recurring pattern, we only push the next instance

#### Editing an existing meeting series
- A meeting series can be canceled which removes it from the meetings view. New meetings will not be created from that series.
- Once a series has been created
  - Team, type, and icebreaker options are fixed
  - What can be edited during an open meeting? Meeting name, calendar/recurring options (just changes the date for the existing)
  - What can be edited only for the next meeting and moving forward? Choose a template

#### Break it down
- A first version of the meetings view with cards ships #4624 
- A meeting can [optionally open the next meeting as a selected default](https://www.figma.com/file/dfTQxhoKzohkCpxlztBbhS/Meetings-view?node-id=103%3A68682). These can be canceled from the meeting card. We’ll probably skip this iteration.
- A meeting can have a date/time. I can’t imagine we’ll build this and not the recurring pattern.
- A meeting can have a recurring pattern. See the specs above
- When creating or editing a date/time and/or recurring pattern, we push the next event to Google Calendar

#### TL;DR
- A series automatically opens the next meeting
- Scheduling information including date, time, and recurring patterns is just editable meta
- The next meeting can be blocked by an administrative action e.g. choose a template
- When scheduling information is added or edited it can be pushed to Google Calendar

### Estimated effort
- 60 hours, break this down

**Comment by ackernaut** [comment]
‘Push next meeting to my calendar’

The idea here is that we create an event on the user’s calendar once folks have dialed in scheduling info on our meeting object. To keep a first version simple (1-way) and avoid cluttering a user’s calendar with obsolete events, we discussed only pushing the next instance when the next meeting in the series is automatically started. E.g.

- Terry creates a retro series starting Fri Jan 29 that recurs every 2 weeks. He checks to push to Google Calendar. He creates the meeting. The first meeting of the series is visible in Parabol. A calendar event is created in Google.
- On Fri Jan 29 Terry ends the retro. The next retro is automatically opened with a calendar date of Fri Feb 12 following the current recurring pattern. An event for that date/time is added to Google Calendar for integrated users.

There are a few problems since we are only creating one event at a time and it’s a 1-way push.

- I put the wrong time and want to update the scheduling. Do I do this on the Parabol meeting? Does that push a new event?
- For some reason I test the meeting and end it early, do I start it again? Does this put the next possible date on the meeting based on the recurring pattern? Is it the same date as it was? Does another event get created on the calendar?
- Who is integrated? If it pushes to my calendar, I can then invite others to that calendar event? If several teammates integrated with Google Calendar in Parabol then they could be automatically invited to the same event?

Some bad ideas:
- If scheduling has been added and pushed to the calendar, but not edited on the series, we only push 1 calendar event per scheduled recurring instance. We check the timestamp of when the meetings in the series end and see if they fall before the next scheduled time. E.g. I set Fri Feb 12, and then I started and stopped 3x before that date. I didn’t change the scheduling, the end times are before that date, I don’t push another calendar event
- If scheduling has been added, a first event pushed to calendar, and then there are edits. Could we integrate a minimal way to detect the old item and remove or update it short of 2-way sync (meaning items changed in Calendar won’t come back to us)? Do we push a new event and let the old event hang?

Just capturing this thinking as I start to think of edge cases outside of the intent. I have what I need to design for intent and we can decide in implementation how to handle cases outside of that intent.

**Comment by ackernaut** [comment]
If we really wanted to break down the work we could do something simple:
- A meeting can optionally be recurring; recurring is the default
- A recurring meeting:
  - Automatically opens the next meeting when a meeting is closed
  - The meeting link always points to the right place (and can be added to many calendar events with confidence)
  - The invite link is still its own thing?
  - It can be canceled from the meetings view
- New users automatically get a recurring meeting of each type on their meetings view; they’d have to cancel them to remove them

**Comment by ackernaut** [comment]
I’ve chipped away at the big, first comment. It’s too bloated **Proposal**:

- All meetings are a series
- The automatically open the next in series until canceled (there’s no option for folks)
- The name has a rolling count until edited, then the name persists, can be edited from the card or meeting UI, the summary has the name as it was for that instance
- They have a clear permalink that can be used on a calendar ad nauseam
- They have a clear invite link
- We offer an affordance to ‘push to Google Calendar’ once from the new meeting view, it just opens a new event with meeting name, a pithy description, and a link back. There, folks can manage date, time, recurring, invitees, add a video conferencing link, etc.
- We don’t keep date, time, recurring meta on the calendar. Calendars are the source of truth. Let’s build other things and provide permalinks to our things, and leak those things into other systems in the right place with a clear link back to close the loop.

**Comment by ackernaut** [comment]
>I’ve chipped away at the big, first comment. It’s too bloated Proposal:
@mattkrick thoughts?

**Comment by mattkrick** [comment]
> New folks who sign up (patient 0, not invitees) get a meeting series of each type automatically on their dashboard view

this might be overwhelming to create an account & all the sudden see you've got 3 meetings that are going on that you didn't ask for. Maybe we just make it extra easy to start a meeting, like make the CTA louder?

> The name has a rolling count until edited,

we may be able to use this as a light form of gamification. what if the number was big & overlaid on the meeting image? sure, they could change the title, but the number would always be somewhere on it. Folks like to see numbers grow. Even if they've been meeting for years, starting at 1 when they join parabol should drive attachment to the product.

**Comment by ackernaut** [comment]
@mattkrick I drafted several issues (breaking down the work), trying to resolve the latest, but there may be incongruence between the issues’ AC (I can scrub further). Usually writing down AC forces us to think about the details and push back:
 
#4623
#4624
#4634
#4636
#4637
#4638
#4639
